### DISCUSSION TITLE
Loading from a .zip file

### DISCUSSION THREAD

**Discussion by ZXMushroom63** [gsuuon/ad-llama#7] [Ideas]
Hello, I was trying to use this library to make a fully offline AI chat app type thing, but after poring through your code, I can't figure out how to use JSZip instead of just making network requests.

Would it be possible to also offer loading from a .zip file?

**Comment by gsuuon** [comment]
You could download the model weights + configs and serve them yourself along with the web app - you would just pass in a [`ModelSpec`](https://gsuuon.github.io/ad-llama/types/_internal_.ModelSpec.html) to [`loadModel`](https://gsuuon.github.io/ad-llama/functions/loadModel.html) which points to the localhost url's. `modelWeightsConfigUrl` needs to point to a directory path that looks like [this](https://huggingface.co/mlc-ai/mlc-chat-Llama-2-7b-chat-hf-q4f32_1/tree/main)

**Comment by ZXMushroom63** [comment]
got it, thanks
